src/Features/Core/Portable/Common/AbstractProjectExtensionProvider.cs

```diff
@@ -87,48 +88,52 @@ public static ImmutableArray<TExtension> GetExtensions(TextDocument document, Fu
 
     public static ImmutableArray<TExtension> FilterExtensions(TextDocument document, ImmutableArray<TExtension> extensions, Func<TExportAttribute, ExtensionInfo> getExtensionInfoForFiltering)
     {
-        return extensions.WhereAsArray(ShouldIncludeExtension);
+        return extensions.WhereAsArray(ShouldIncludeExtension, (document, getExtensionInfoForFiltering));
 
-        bool ShouldIncludeExtension(TExtension extension)
+        static bool ShouldIncludeExtension(TExtension extension, (TextDocument, Func<TExportAttribute, ExtensionInfo>) args)
```

Comment:
i have no problem with you inlining this method into WhereAsArray.

Reply:
Tried it, stared at it for a minute, and I personally like it better in the current form. Will gladly change it though if you feel fairly strongly or have an objective reason.